solr/core/src/java/org/apache/solr/cluster/events/impl/ClusterEventProducerFactory.java

```diff
@@ -128,13 +128,17 @@ public void added(ContainerPluginsRegistry.ApiInfo plugin) {
           ClusterEventListener listener = (ClusterEventListener) instance;
           clusterEventProducer.registerListener(listener);
         } else if (instance instanceof ClusterEventProducer) {
-          // replace the existing impl
-          if (cc.getClusterEventProducer() instanceof DelegatingClusterEventProducer) {
-            ((DelegatingClusterEventProducer) cc.getClusterEventProducer())
-                .setDelegate((ClusterEventProducer) instance);
+          if (ClusterEventProducer.PLUGIN_NAME.equals(plugin.getInfo().name)) {
+            // replace the existing impl
+            if (cc.getClusterEventProducer() instanceof DelegatingClusterEventProducer) {
+              ((DelegatingClusterEventProducer) cc.getClusterEventProducer())
+                  .setDelegate((ClusterEventProducer) instance);
```

Comment:
It seems we still have synchronization issues on access to `DelegatingClusterEventProducer.delegate`, right? Might not be on this PR though (and I might have it wrong because I don't fully understand this plugin and events architecture) but in `DelegatingClusterEventProducer` I see non sync access to `delegate` and I wonder...